RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -0,0 +1,277 @@
+using System.Collections.Immutable;
+using System.Data;
+
+namespace RubeesEat.Model.DB;
+
+public class DbBillRepository(IDbConnectionFactory connectionFactory) : IBillRepository
+{
+    public IReadOnlyList<Bill> GetAll()
+    {
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "SELECT BillID, Date, Description FROM Bills ORDER BY Date DESC;";
+        command.CommandType = CommandType.Text;
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+        var bills = new List<Bill>();
+        while (reader.Read())
+        {
+            var guid = reader.GetGuid(0);
+            bills.Add(new Bill
+            (
+                guid,
+                reader.GetDateTime(1),
+                reader.GetString(2),
+                [..GetEntryLines(guid)]
+            ));
+        }
+
+        return bills.AsReadOnly();
+    }
+
+    private EntryLine[] GetEntryLines(Guid billID)
+    {
+        var entryLines = new List<EntryLine>();
+
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "SELECT PersonID, Amount FROM EntryLines WHERE BillID = @BillID;";
+        command.CommandType = CommandType.Text;
+
+        var billIdParam = command.CreateParameter();
+        billIdParam.ParameterName = "@BillID";
+        billIdParam.Value = billID;
+        command.Parameters.Add(billIdParam);
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+
+        while (reader.Read())
+        {
+            var guid = reader.GetGuid(0);
+            entryLines.Add(new EntryLine(
+                GetPersonById(guid),
+                reader.GetDecimal(1)
+            ));
+        }
+
+        return entryLines.ToArray();
+
+
+        Person GetPersonById(Guid guid)
+        {
+            var personRepository = new DbPersonRepository(connectionFactory);
+            return personRepository.GetById(guid);
+        }
```

Comment:
Same thing as above. In addition to retrieving all the entry lines we also retrieve all the persons for each entry line. If a Bill has two entry lines and we request 100 bills we make a total of 301 requests to the server. This is unacceptable as it will slow down everything drastically. At minimum, we should only requests the same person once.